### PR TITLE
Fix incorrect result of sql server tinyint

### LINKED ISSUE #11209: SqlServer TINYINT incorrect results
We mapped Trino `TINYINT` to SqlServer `TINYINT`, but I found that the mapping relationship is **wrong**, why?

- In Trino, the range of `TINYINT` is `[-128, 127]`
- In SqlServer, the range of `TINYINT` is `[0, 255]`

So we **can't** read or write `TINYINT` values of `[128, 255]` through Trino.

Assuming I have a table `test_table (row_id integer, data tinyint)` in SqlServer, and the following data in the table:

```
1 127
2 128
```

Then, if I read it through Trino, I will get the following result:

```
1 127
2 -128 <- error value
```

Since Trino's `TINYINT` supports negative numbers, what happens if I write a negative number to SqlServer via Trino?  (We can't write negative numbers via SqlServer JDBC `Statement`)

```
INSERT INTO test_table VALUES (3, -1)
```

query via SqlServer JDBC:

```
1 127
2 128
3 255 <- error value
```

query via Trino:

```
1 127
2 -128
3 -1 <- error value
```

### PR DESCRIPTION
**Pull request by tangjiangling**
## Description

> Is this change a fix, improvement, new feature, refactoring, or other?

Fix the correctness of query results.

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

SQL Server connector.

> How would you describe this change to a non-technical end user or system administrator?

Fix incorrect result of SQL Server TINYINT.

## Related issues, pull requests, and links

Fixes #11209

## Documentation

(x) Sufficient documentation is included in this PR.

## Release notes

(x) Release notes entries required with the following suggested text:

```markdown
# SQL Server
* Fix incorrect results when querying SQL Server ``tinyint`` columns by mapping them to Trino ``smallint``. ({issue}`11209`)
```

### CODE REVIEW

File: plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/AbstractTestOracleTypeMapping.java [hunk 1]
```diff
@@ -106,7 +106,7 @@ public void setUp()
 
     private DataSetup trinoCreateAsSelect(String tableNamePrefix)
     {
-        return new CreateAsSelectDataSetup(new TrinoSqlExecutor(getQueryRunner()), tableNamePrefix);
```
Comment (hashhar) [inline plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/AbstractTestOracleTypeMapping.java:109]: Commit message could be a one-line "Reuse trinoCreateAndInsert and trinoCreateAsSelect in type mapping tests".

File: plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java [hunk 2]
```diff
@@ -285,7 +284,9 @@ protected void copyTableSchema(Connection connection, String catalogName, String
                 return Optional.of(booleanColumnMapping());
 
             case Types.TINYINT:
-                return Optional.of(tinyintColumnMapping());
+                // Map SQL Server TINYINT to Trino SMALLINT because SQL Server TINYINT is actually "unsigned tinyint"
+                // We don't check the range of values, because SQL Server will do it for us
```
Comment (hashhar) [inline plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java:288]: also mention the test which covers this

**Review by hashhar — APPROVED** [review]

**Comment by hashhar** [comment]
docs need to be updated in `sqlserver.rst` type mapping table

**Comment by tangjiangling** [comment]
I found that there are some other type mappings not covered by the SQL Server documentation (e.g. `bit`, `decimal`, `tinyint`, etc.), so I will only add `tinyint` related documentation in this PR, and I will file an issue to track the remaining parts.

File: docs/src/main/sphinx/connector/sqlserver.rst [hunk 3]
```diff
@@ -127,6 +127,7 @@ SQL Server Type                     Trino Type
 ``bigint``                          ``bigint``
 ``smallint``                        ``smallint``
 ``int``                             ``integer``
+``tinyint``                         ``smallint``
```
Comment (hashhar) [inline docs/src/main/sphinx/connector/sqlserver.rst:130]: This should come before `smallint`.

Comment (tangjiangling) [reply docs/src/main/sphinx/connector/sqlserver.rst:130]: I have a question about the type mapping documentation: what is the principle of ordering, I checked some other connector's documentation and there seems to be no uniform principle.

One suggestion is: could we be consistent with the type order defined by `java.sql.Types`, I observed that the order of the types in `JdbcClient#toColumnMapping` would be consistent with the order defined in `java.sql.Types`.

**Review by hashhar — APPROVED** [review]
LGTM.

Thanks a lot @tangjiangling.

**Comment by hashhar** [comment]
@tangjiangling CI is failing. You'll also need to adjust `io.trino.tests.product.sqlserver.TestSelect.testAllDatatypes` it seems.

**Comment by tangjiangling** [comment]
Got it, I'll take a look.

**Comment by tangjiangling** [comment]
Updated and successfully ran SQL Server PTs locally.

**Pull request merged**